components/intl_provider/intl_provider.tsx

```diff
@@ -1,27 +1,30 @@
 // Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
 // See LICENSE.txt for license information.
 
-import PropTypes from 'prop-types';
+import {ReactElementLike} from 'prop-types';
 import React from 'react';
 import {IntlProvider as BaseIntlProvider} from 'react-intl';
 
+import {MessageFormatElement} from '@formatjs/icu-messageformat-parser';
+
 import {Client4} from 'mattermost-redux/client';
 import {setLocalizeFunction} from 'mattermost-redux/utils/i18n_utils';
 
 import * as I18n from 'i18n/i18n';
 
 import {localizeMessage} from 'utils/utils';
-
-export default class IntlProvider extends React.PureComponent {
-    static propTypes = {
-        children: PropTypes.element.isRequired,
-        locale: PropTypes.string.isRequired,
-        translations: PropTypes.object,
-        actions: PropTypes.shape({
-            loadTranslations: PropTypes.func.isRequired,
-        }).isRequired,
+import {ActionFunc} from 'mattermost-redux/types/actions';
+
+type Props = {
+    children: ReactElementLike;
+    locale: string;
+    translations: Record<string, string> | Record<string, MessageFormatElement[]> | null;
```

Comment:
I don't think we explicitly check for `null`, can we instead make this optional? (ie. `undefined` instead of `null`)
```suggestion
    translations?: Record<string, string> | Record<string, MessageFormatElement[]>;
```

Reply:
In associated test, we have scenarii that use explicitly `translations` with `null` values, that's why I kept `null`. I could replace tests with `null` by tests with `undefined` (tests pass), if you are sure it's ok.

Reply:
I gave a try in commit 344a76a, please tell me if you feel it convenient